data_model/primitives/src/lib.rs

```diff
@@ -15,3 +15,91 @@ extern crate alloc;
 pub mod atomic;
 pub mod fixed;
 pub mod small;
+
+use fixed::prelude::*;
+
+/// Trait for values that could be converted to `f64` and observed by prometheus.
+pub trait FloatMetricFrom {
+    /// Convert `value` to `f64`.
+    fn from(value: Self) -> f64;
+}
+
+impl FloatMetricFrom for u128 {
+    #[inline]
+    #[allow(clippy::cast_precision_loss)]
+    fn from(value: Self) -> f64 {
+        value as f64
+    }
+}
+
+impl FloatMetricFrom for u32 {
+    #[inline]
+    fn from(value: Self) -> f64 {
+        value.into()
+    }
+}
+
+impl FloatMetricFrom for Fixed {
+    #[inline]
+    fn from(value: Self) -> f64 {
+        value.into()
+    }
+}
+
+/// Trait for `checked_add` operation on iroha's primitive types.
+pub trait CheckedAdd {
```

Comment:
I think checked operations need to be tied together into a (single) trait. It's unlike that we'll need only one of those two.